core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushProjectionThroughJoin.java

```diff
@@ -69,11 +69,11 @@ public static Optional<PlanNode> pushProjectionThroughJoin(
         for (Map.Entry<Symbol, Expression> assignment : projectNode.getAssignments().entrySet()) {
             Expression expression = assignment.getValue();
             Set<Symbol> symbols = extractUnique(expression);
-            if (leftChild.getOutputSymbols().containsAll(symbols)) {
+            if (ImmutableSet.copyOf(leftChild.getOutputSymbols()).containsAll(symbols)) {
```

Comment:
I think it would be nice to handle in https://github.com/trinodb/trino/pull/19260

Reply:
Dropped. I've fixed that because I was in that class and spotted it.